### PR TITLE
Removing RobertaConfig inheritance from CamembertConfig

### PR DESCRIPTION
**Pull request by Saad135**
# What does this PR do?

Removes RobertaConfig dependencies from CamembertConfig

Related to https://github.com/huggingface/transformers/issues/19303

@sgugger can I please get some feedback on this. Thanks 😄 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/camembert/configuration_camembert.py [hunk 1]
```diff
@@ -36,12 +36,57 @@
...
     Camembert [camembert-base](https://huggingface.co/camembert-base) architecture.
 
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        vocab_size (`int`, *optional*, defaults to 30522):
+            Vocabulary size of the BERT model. Defines the number of different tokens that can be represented by the
+            `inputs_ids` passed when calling [`BertModel`] or [`TFBertModel`].
```
Comment (sgugger) [inline src/transformers/models/camembert/configuration_camembert.py:53]: Should be
```suggestion
            Vocabulary size of the BERT model. Defines the number of different tokens that can be represented by the
            `inputs_ids` passed when calling [`CamembertModel`] or [`TFCamembertModel`].
```

File: src/transformers/models/camembert/configuration_camembert.py [hunk 2]
```diff
@@ -36,12 +36,57 @@
...
+            `"relu"`, `"silu"` and `"gelu_new"` are supported.
+        hidden_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout probability for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        max_position_embeddings (`int`, *optional*, defaults to 512):
+            The maximum sequence length that this model might ever be used with. Typically set this to something large
+            just in case (e.g., 512 or 1024 or 2048).
+        type_vocab_size (`int`, *optional*, defaults to 2):
+            The vocabulary size of the `token_type_ids` passed when calling [`BertModel`] or [`TFBertModel`].
```
Comment (sgugger) [inline src/transformers/models/camembert/configuration_camembert.py:73]:
```suggestion
            The vocabulary size of the `token_type_ids` passed when calling [`CamembertModel`] or [`TFCamembertModel`].
```

File: src/transformers/models/camembert/configuration_camembert.py [hunk 3]
```diff
@@ -36,12 +36,57 @@
 }
 
 
-class CamembertConfig(RobertaConfig):
+class CamembertConfig(PretrainedConfig):
     """
-    This class overrides [`RobertaConfig`]. Please check the superclass for the appropriate documentation alongside
-    usage examples. Instantiating a configuration with the defaults will yield a similar configuration to that of the
+    This is the configuration class to store the configuration of a [`CamembertConfig`] or a [`TFCamembertConfig`]. It is
```
Comment (sgugger) [inline src/transformers/models/camembert/configuration_camembert.py:41]:
```suggestion
    This is the configuration class to store the configuration of a [`CamembertModel`] or a [`TFCamembertModel`]. It is
```

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! There are just a couple of copy-paste to adapt. Also make sure you run `make style` on your branch to fix the formatting.

**Comment by Saad135** [comment]
> Also make sure you run `make style` on your branch to fix the formatting.

`make style` changed some files in other folders as well related to other models. Since I didn't change them, I didn't add them in this PR, and only added the camembert_configuration.py file with the fixed style.

I was unsure if I should add style changes in other files in this PR, since this PR is about CamembertConfig.

File: src/transformers/models/camembert/configuration_camembert.py (hunk 1 above)
Comment (Saad135) [reply src/transformers/models/camembert/configuration_camembert.py]: > Vocabulary size of the BERT model.

Should the BERT her be changed to Camembert? (Line 52)

**Review by sgugger — APPROVED** [review]
It may be because you have a slightly different version of black and not 22.3
You did well by not including the other changes :-)

**Pull request merged**